File: haas_lib_bundles/python/libraries/dcmotor/dcmotor.py
from driver import GPIO

class DCMOTOR(object):

    def __init__(self, gpioObj):
        self.gpioObj = None
        if not isinstance(gpioObj, GPIO):
            raise ValueError("parameter is not a GPIO object")
        self.gpioObj = gpioObj

    def ctrl(self,value):
        if self.gpioObj is None:
            raise ValueError("invalid GPIO object")
        self.gpioObj.write(value)

File: components/amp/example-js/ESP32/jsapi/gpio/dcmotor/dcmotor.js
class _DCMOTOR {
    constructor(gpioObj) {
        this.gpioInstance = gpioObj;
    };

    ctrl(onoff) {
	this.gpioInstance.writeValue(onoff);
    };
}

export function DCMOTOR(gpioObj) {
    return new _DCMOTOR(gpioObj);
}
